src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceIntellisense/IIntellisenseClient.cs

```diff
@@ -0,0 +1,14 @@
+using Kusto.Language;
+
+namespace Microsoft.Kusto.ServiceLayer.DataSource.DataSourceIntellisense
+{
+    public interface IIntellisenseClient
```

Comment:
Created an interface for a client to handle creating and persisting intellisense.